packages/material-ui/src/ObjectFieldTemplate/ObjectFieldTemplate.tsx

```diff
@@ -49,16 +52,19 @@ const ObjectFieldTemplate = ({
         />
       )}
       <Grid container={true} spacing={2} className={classes.root}>
-        {properties.map((element: any, index: number) => (
-          <Grid
-            item={true}
-            xs={12}
-            key={index}
-            style={{ marginBottom: '10px' }}
-          >
-            {element.content}
-          </Grid>
-        ))}
+        {properties.map((element: any, index: number) =>
+          isHidden(element) ? (
```

Comment:
Why do we need this change in ObjectFieldTemplate -- what does it do? Isn't the FieldTemplate change enough?

Reply:
Here (`ObjectFieldTemplate`) a `<Grid>` with a margin is added, which means that a hidden element would still take up space. We have to remove the `<Grid>` if the field is hidden.

Reply:
Do you think the code needs a comment to explain this?

Reply:
Oh, I see. Instead of doing this, could we check if `element.content === null` and then remove the grid? I'm asking because it's currently possible to specify widgets in different ways than `ui:widget` (we can do `{'ui:options': {'widget': ...`) so it's good not to tie the implementation specifically to `ui:widget`.

Can you also add a comment describing why we're hiding the grid in this case?

Reply:
hmm `element.content === null` is not `true` in that case. There is still a field create, just a hidden one. Is there any other way I can use to support both cases without having to hard code it in that spot?

Reply:
What is element.content equal to in that case?

Reply:
```typescript
{
            content: React.ReactElement;
            name: string;
            disabled: boolean;
            readonly: boolean;
}
```
https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/index.d.ts#L213-L218 

If `@rjsf/core` would add a `hidden` property there too it would be an easy.

Reply:
Oh, just noticed that you ask for `.content`...  Not 100% sure, but I think it is a `SchemaField` from `core`.

Reply:
But in that case they would both evaluate the same condition: https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/src/components/fields/SchemaField.js#L307

Reply:
Yeah, good point.

I wonder if we could still pass the `hidden` prop over to ObjectFieldTemplate from `@rjsf/core`, though, as this would reduce code duplication with checking ui:widget. How feasible / easy do you think that change would be?

Reply:
Doing this would also help out with https://github.com/rjsf-team/react-jsonschema-form/pull/2175! (similar issue with bootstrap-4 theme)

Reply:
Had some trouble with `nanoid`, but I think I got it working now. Still have to postpone it till tomorrow.

Reply:
PR is updated. I couldn't find a good place in `core` to test it, so I skipped that. I replaced the old implementation in `antd`. I also went through all themes and fixed #2175 for bootstrap-4 and fluid-ui.